presto-parser/src/main/java/com/facebook/presto/sql/tree/AlterColumnNotNull.java

```diff
@@ -0,0 +1,121 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class AlterColumnNotNull
+        extends Statement
+{
+    private final QualifiedName table;
+    private final Identifier column;
+    private final boolean tableExists;
+    private final boolean columnExists;
+    private final boolean dropConstraint;
+
+    public AlterColumnNotNull(QualifiedName table, Identifier column, boolean tableExists, boolean columnExists, boolean dropConstraint)
+    {
+        this(Optional.empty(), table, column, tableExists, columnExists, dropConstraint);
+    }
+
+    public AlterColumnNotNull(NodeLocation location, QualifiedName table, Identifier column, boolean tableExists, boolean columnExists, boolean dropConstraint)
+    {
+        this(Optional.of(location), table, column, tableExists, columnExists, dropConstraint);
+    }
+
+    private AlterColumnNotNull(Optional<NodeLocation> location, QualifiedName table, Identifier column, boolean tableExists, boolean columnExists, boolean dropConstraint)
+    {
+        super(location);
+        this.table = requireNonNull(table, "table is null");
+        this.column = requireNonNull(column, "column is null");
+        this.tableExists = tableExists;
+        this.columnExists = columnExists;
+        this.dropConstraint = dropConstraint;
+    }
+
+    public QualifiedName getTable()
+    {
+        return table;
+    }
+
+    public Identifier getColumn()
+    {
+        return column;
+    }
+
+    public boolean isTableExists()
+    {
+        return tableExists;
+    }
+
+    public boolean isColumnExists()
+    {
+        return columnExists;
+    }
+
+    public boolean isDropConstraint()
+    {
+        return dropConstraint;
+    }
+
+    @Override
+    public <R, C> R accept(AstVisitor<R, C> visitor, C context)
+    {
+        return visitor.visitAlterColumnNotNull(this, context);
+    }
+
+    @Override
+    public List<Node> getChildren()
+    {
+        return ImmutableList.of();
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        AlterColumnNotNull that = (AlterColumnNotNull) o;
+        return Objects.equals(table, that.table) &&
+                Objects.equals(column, that.column) &&
+                Objects.equals(dropConstraint, that.dropConstraint);
```

Comment:
Have you intentionally left out `tableExists` and `columnExists`?

Reply:
Yes. I'm not quite sure where this comparison code is called, but we always seem to omit the exists flags. See RenameColumn, AddColumn, RenameTable, etc.

Reply:
@ClarenceThreepwood  it looks like you added these exists fields to those classes https://github.com/prestodb/presto/pull/15003 (via cherry-picking a trino commit). Are you sure it wasn't just an oversight?

Reply:
That is true, and it was probably an oversight 4 years ago. Do you think these fields should be added? I can add these for all the relevant parse nodes if so

Reply:
Added